### PR TITLE
[11.0][mig][account_fiscal_year_closing] WIP

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
WIP. Seems that issue https://github.com/odoo/odoo/issues/20163 was not completely resolved. I am asking to reopen.

### CODE REVIEW

**Comment by JordiBForgeFlow** [comment]
@pedrobaeza Seems to me that the bug odoo/odoo#20163 will not have an easy fix. I propose to continue with the approach of v9 and v10.

**Comment by pedrobaeza** [comment]
Well, as said by them, the workaround should work, so let's give them a bit more time... If not , we can always easily switch back to the old mechanism.

File: account_fiscal_year_closing/models/account_fiscalyear_closing.py [hunk 1]
```diff
@@ -0,0 +1,641 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Tecnativa - Antonio Espinosa
+# Copyright 2017 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+from openerp import _, api, exceptions, fields, models
```
Comment (zamberjo) [inline account_fiscal_year_closing/models/account_fiscalyear_closing.py]: Maybe you can replace all `from openerp ...` to `from odoo ...`

Or execute `find . -type f -name '*.py' | xargs sed -i 's/from openerp/from odoo/g'` on root.

File: account_fiscal_year_closing/views/account_fiscalyear_closing_template_views.xml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                    <group>
+                        <group>
+                            <field name="check_draft_moves"/>
+                        </group>
+                        <group>
+                            <field name="chart_template_ids" widget="many2many_tags"/>
+                            <field name="company_id" groups="base.group_multi_company"/>
+                        </group>
+                    </group>
+                    <group string="Moves configuration">
```
Comment (zamberjo) [inline account_fiscal_year_closing/views/account_fiscalyear_closing_template_views.xml]: Replace string selectors by name. 

`<group name="Moves configuration">`

File: account_fiscal_year_closing/views/account_fiscalyear_closing_template_views.xml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                                            <field name="sequence"/>
+                                        </group>
+                                        <group>
+                                            <field name="code"/>
+                                            <field name="inverse"/>
+                                            <field name="reconcile"/>
+                                            <field name="move_type"/>
+                                        </group>
+                                    </group>
+                                    <group string="Accounts mapping">
```
Comment (zamberjo) [inline account_fiscal_year_closing/views/account_fiscalyear_closing_template_views.xml]: Replace string selectors by name.

File: account_fiscal_year_closing/views/account_fiscalyear_closing_template_views.xml [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+                                        <field name="mapping_ids" nolabel="1">
+                                            <tree editable="bottom">
+                                                <field name="name"/>
+                                                <field name="src_accounts"/>
+                                                <field name="dest_account"/>
+                                            </tree>
+                                        </field>
+                                    </group>
+                                    <group string="Accounts closing types">
```
Comment (zamberjo) [inline account_fiscal_year_closing/views/account_fiscalyear_closing_template_views.xml]: Replace string selectors by name.

File: account_fiscal_year_closing/views/account_fiscalyear_closing_views.xml [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+                            <field name="date_opening"/>
+                            <field name="check_draft_moves"/>
+                        </group>
+                    </group>
+                    <group attrs="{'invisible': [('calculation_date', '=', False)]}">
+                        <group>
+                            <field name="calculation_date"/>
+                        </group>
+                    </group>
+                    <group string="Moves configuration">
```
Comment (zamberjo) [inline account_fiscal_year_closing/views/account_fiscalyear_closing_views.xml]: Replace string selectors by name.

File: account_fiscal_year_closing/views/account_fiscalyear_closing_views.xml [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+                                            <field name="sequence"/>
+                                        </group>
+                                        <group>
+                                            <field name="code"/>
+                                            <field name="inverse"/>
+                                            <field name="reconcile"/>
+                                            <field name="move_type"/>
+                                        </group>
+                                    </group>
+                                    <group string="Accounts mapping">
```
Comment (zamberjo) [inline account_fiscal_year_closing/views/account_fiscalyear_closing_views.xml]: Replace string selectors by name.

File: account_fiscal_year_closing/views/account_fiscalyear_closing_views.xml [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+                                        <field name="mapping_ids" nolabel="1">
+                                            <tree editable="bottom">
+                                                <field name="name"/>
+                                                <field name="src_accounts"/>
+                                                <field name="dest_account_id"/>
+                                            </tree>
+                                        </field>
+                                    </group>
+                                    <group string="Accounts closing types">
```
Comment (zamberjo) [inline account_fiscal_year_closing/views/account_fiscalyear_closing_views.xml]: Replace string selectors by name.

**Review by zamberjo — APPROVED** [review]

File: account_fiscal_year_closing/views/account_fiscalyear_closing_template_views.xml (hunk 2 above)
Comment (pedrobaeza) [reply account_fiscal_year_closing/views/account_fiscalyear_closing_template_views.xml]: It shouldn't be replaced, but complemented:
```
<group string="Moves configuration" name="moves_configuration"/>
```

File: account_fiscal_year_closing/__init__.py [hunk 8]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
```
Comment (zamberjo) [inline account_fiscal_year_closing/__init__.py:1]: Not needed in Python3

File: account_fiscal_year_closing/__manifest__.py [hunk 9]
```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
```
Comment (zamberjo) [inline account_fiscal_year_closing/__manifest__.py:1]: Not needed in Python3

File: account_fiscal_year_closing/models/__init__.py [hunk 10]
```diff
@@ -0,0 +1,6 @@
+# -*- coding: utf-8 -*-
```
Comment (zamberjo) [inline account_fiscal_year_closing/models/__init__.py:1]: Not needed in Python3

File: account_fiscal_year_closing/models/account_fiscalyear_closing_abstract.py [hunk 11]
```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
```
Comment (zamberjo) [inline account_fiscal_year_closing/models/account_fiscalyear_closing_abstract.py:1]: Not needed in Python3

File: account_fiscal_year_closing/models/account_fiscalyear_closing_template.py [hunk 12]
```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
```
Comment (zamberjo) [inline account_fiscal_year_closing/models/account_fiscalyear_closing_template.py:1]: Not needed in Python3

File: account_fiscal_year_closing/models/account_move.py [hunk 13]
```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
```
Comment (zamberjo) [inline account_fiscal_year_closing/models/account_move.py:1]: Not needed in Python3

**Review by zamberjo — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Closing as no need of this one finally.

**Comment by sergiocorato** [comment]
@pedrobaeza can you reopen this as of thread Functional Question: Year End Closing in Odoo (OPEJ) in mailing list? I based the v. 12.0 on this one.

**Comment by pedrobaeza** [comment]
No, you should do a new PR based on this PR, as the original contributor is not interested anymore in it.

**Comment by sergiocorato** [comment]
ah ok

**Comment by eng-eslam-a-y** [comment]
> Closing as no need of this one finally.

@pedrobaeza 
Hi, could you please tell me why this one is closed as i didn't find a reason why.
if there is workaround for it or another OCA module in v12 could you please guide me to it

**Comment by pedrobaeza** [comment]
No need of fiscal year closing since v9.

**Comment by sergiocorato** [comment]
@pedrobaeza as of your reply on contributor mailing list I opened #129 (and #134) in this repo, if preferable I'll move it on Italian localization.

**Comment by pedrobaeza** [comment]
Yes, but the general reply still is that is not needed. You insist on putting it, then you continue with the PR, but better to not confuse others believing that the module is needed in a general way.

**Comment by sergiocorato** [comment]
I agree, this module is useful only if user is absolutely sure that laws in his country require it.

**Pull request closed without merging**